### PR TITLE
Fixed suppressed class member comments

### PR DESCRIPTION
**Pull request by cmichaelgraham**
Also added tests for commented class members

[Review on Reviewable](https://reviewable.io/reviews/yolodev/babel-dts-generator/38)

### CODE REVIEW

**Comment by Alxandr** [comment]
Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

_[spec/comments/comment-class-method.expected.ts, line 12 [r1]](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38#-KBbnWCTR9lYCoS7InAS:-KBbnWCUNfobetmf2oPD:-418272256) ([raw file](https://github.com/yolodev/babel-dts-generator/blob/fb92f2aae4c57de7a08936e07cdcb38d900341ec/spec/comments/comment-class-method.expected.ts#L12)):_
Missing blank line?

---

_[src/generators/ast.js, line 557 [r1]](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38#-KBbnlgk5hHo7O12mPqH:-KBbnlgk5hHo7O12mPqI:-2062425750) ([raw file](https://github.com/yolodev/babel-dts-generator/blob/fb92f2aae4c57de7a08936e07cdcb38d900341ec/src/generators/ast.js#L557)):_
Hold on, this is definitely not the right place to do this. `toCode` is a convenience method so you don't have to do `node => node.toCode(ctx)` everywhere you use `Array.prototype.map` to generate code. Find out where you need these, don't just apply them globally.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38)_

**Comment by cmichaelgraham** [comment]
Review status: 6 of 7 files reviewed at latest revision, 2 unresolved discussions.

---

_[spec/comments/comment-class-method.expected.ts, line 12 [r1]](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38#-KBbnWCTR9lYCoS7InAS:-KBcJ3Mtl-wxVdFH6o47:-499146009) ([raw file](https://github.com/yolodev/babel-dts-generator/blob/fb92f2aae4c57de7a08936e07cdcb38d900341ec/spec/comments/comment-class-method.expected.ts#L12)):_
Done.

---

_[src/generators/ast.js, line 557 [r1]](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38#-KBbnlgk5hHo7O12mPqH:-KBcIa2CA6x412JQHtR4:-2001887603) ([raw file](https://github.com/yolodev/babel-dts-generator/blob/fb92f2aae4c57de7a08936e07cdcb38d900341ec/src/generators/ast.js#L557)):_
I might need some help with this one.  I added the `includeComment` and `includeCode` properties because the comments were being suppressed in some cases.  I'm not sure I understand how to make the change you are requesting...

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38)_

**Comment by Alxandr** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_[src/generators/ast.js, line 557 [r1]](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38#-KBbnlgk5hHo7O12mPqH:-KBcdIjlga8Q6zGvov4w:-481703072) ([raw file](https://github.com/yolodev/babel-dts-generator/blob/fb92f2aae4c57de7a08936e07cdcb38d900341ec/src/generators/ast.js#L557)):_
So, the thing is, that this is a recursive algorithm, right? What you're doing is just blatantly turning on comments _everywhere_, independently on basically everything. This is again caused by the way you turn `includeComments` and `includeCode` to false in some cases. These cases doesn't go just one level deep, and I think that is where your problem is, and that is what needs to be solved.

So, basically, refactor away `includeCode` (which shouldn't be a thing), and this is likely to just resolve itself.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38)_

**Comment by cmichaelgraham** [comment]
Review status: 6 of 7 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_[src/generators/ast.js, line 557 [r1]](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38#-KBbnlgk5hHo7O12mPqH:-KBvsynJk19nL1oBWgNQ:655903992) ([raw file](https://github.com/yolodev/babel-dts-generator/blob/fb92f2aae4c57de7a08936e07cdcb38d900341ec/src/generators/ast.js#L557)):_
Great suggestion.  Please take a look at the refactoring.  All tests are passing and special ctx properties have been removed.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38)_

**Comment by Alxandr** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_[src/generators/ast.js, line 108 [r3]](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38#-KBvu6gfsujFEprqxcfA:-KBvu6gfsujFEprqxcfB:1383618353) ([raw file](https://github.com/yolodev/babel-dts-generator/blob/444bb56753099ab2369e1282cbe57b1059badc88/src/generators/ast.js#L108)):_
Much better. Except, don't use `{ ...ctx }`. That's just needless copying. If you're not going to change anything, simply pass it `ctx` itself. Same for the line below.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38)_

**Comment by cmichaelgraham** [comment]
Review status: 6 of 7 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_[src/generators/ast.js, line 108 [r3]](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38#-KBvu6gfsujFEprqxcfA:-KBvwke1Aug7A_P3c_a7:-499146009) ([raw file](https://github.com/yolodev/babel-dts-generator/blob/444bb56753099ab2369e1282cbe57b1059badc88/src/generators/ast.js#L108)):_
Done.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38)_

**Comment by Alxandr** [comment]
Reviewed 3 of 3 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/yolodev/babel-dts-generator/38)_
